python/ray/actor.py

```diff
@@ -657,6 +662,11 @@ def _remote(self,
             function_signature = meta.method_meta.signatures["__init__"]
             creation_args = signature.flatten_args(function_signature, args,
                                                    kwargs)
+
+        if collective:
+            override_environment_variables = collective_to_envs(
+                collective, override_environment_variables)
```

Comment:
@stephanie-wang or @ericl can you take a quick look and give +1 if ok?

Reply:
This doesn't seem necessary. Can we avoid making changes to core API?

Reply:
@ericl understood the requirement. 

This modification over core APIs was intended to implement an declarative interface: `actor.options(..., collective=options)` per Stephanie's suggestions. See [here](https://gist.github.com/stephanie-wang/ae9a82b3f200989ba37749d3268c0907#file-ray_collective-py-L52)

But I am actually fine not adding this one as we currently have had a declarative interface as `declare_collective_group(actors, options, ...)` in addition to the `init_collective_group(...)` interface.

@stephanie-wang please comment, thanks!